### DISCUSSION TITLE
Reorder Apex test result sections so most important info is at bottom of terminal window

### DISCUSSION THREAD

**Discussion by dschach** [forcedotcom/cli#2872] [Ideas]
As a developer, I want to know easily if ANY of my Apex tests fail when running tests via the VSCode command palette so that I can quickly decide how to proceed.

## Current Output

THe three sections (top to bottom) when invoking Apex tests from the command palette in VSCode:

1. Test Summary
2. List All Tests
3. Apex Code Coverage by Class
This means that I have no quick way to find if the overall outcome is Passed/Failed without a lot of scrolling.

## Suggested Output

Because the output is in a terminal window, the most important information should be at the BOTTOM so there is no need to scroll up. 

TOP: List All Tests
MIDDLE: Apex Code Coverage by Class
BOTTOM (can be seen without scrolling): Test Summary

**Comment by k-capehart** [comment]
@dschach I completely agree. It has always annoyed me that I have to scroll up to the top of the output when I run tests. I was able to make this change here: https://github.com/forcedotcom/salesforcedx-apex/pull/377

Someone from Salesforce can let us know if there are considerations that we haven't though of, since this code is in the apex node library.

**Comment by daveespo** [comment]
IMO, this feature suggestion only addresses a narrow portion of what's problematic with the human-readable output form apex:test:execute -- https://github.com/forcedotcom/salesforcedx-apex/issues/243 has gone without love for a few years now -- I don't see any good reason why thousands of lines of "PASSING" test output should be displayed

**Reply by k-capehart** [reply]
You're right, I came across your issue while I was researching this. It's a bit of a bigger effort, but there might be some ways to improve it incrementally. The entire massive output is almost completely unnecessary.

**Reply by k-capehart** [reply]
@daveespo I've actually just updated the PR to add a parameter to optionally skip outputting the successful tests. I had some time to implement it. If it gets merged, then the next step would be to add a `verbose` (or `concise`) flag to the cli command to pass it on to the format function, whichever gets approved by someone over at Salesforce.

**Reply by daveespo** [reply]
Excellent .. you probably saw I posted over on the https://github.com/forcedotcom/salesforcedx-apex/issues/243 to see if we can get this pain point relieved once and for all. Nice work. :-)

**Comment by k-capehart** [comment]
@dschach This has actually been done now for a couple releases. The latest version of the cli will now display the test result summary at the bottom of the output when running `apex test run`. 

The other topics mentioned are pending a PR merge (https://github.com/salesforcecli/plugin-apex/pull/504) and are being tracked in this issue: https://github.com/forcedotcom/salesforcedx-apex/issues/243

So we're probably free to close this post 🚀
